tests/conftest.py

```diff
@@ -26,11 +26,24 @@ def allowed_audiences() -> List[str]:
 
 
 @pytest.fixture(autouse=True)
-def clear_authz_collector():
+def clear_authz_collector() -> Generator:
     yield
     authz_collector.clean()
 
 
+@pytest.fixture(autouse=True)
+def clear_router_collector() -> Generator:
+    yield
+    Router()._clear()  # pylint: disable=protected-access
+
+
+@pytest.fixture()
+def sample_router() -> Router:
+    router = Router()
+    router.add_route("/", "GET", "x")
```

Comment:
I am a little bit surprised that `handler` is a string 😅 

I think I need to check this code more deeply when I find a moment.